rosbag2_compression/test/rosbag2_compression/test_zstd_compressor.cpp

```diff
@@ -76,13 +94,82 @@ class CompressionHelperFixture : public rosbag2_test_common::TemporaryDirectoryF
 
   void SetUp() override
   {
+    allocator_ = rcutils_get_default_allocator();
+    message_ = create_garbage_string();
+
     rclcpp::init(0, nullptr);
   }
 
   void TearDown() override
   {
     rclcpp::shutdown();
   }
+
+  int write_data_to_serialized_string_message(
+    uint8_t * buffer, size_t buffer_capacity, const std::string & message)
+  {
+    // This function also writes the final null charachter, which is absent in the CDR format.
+    // Here this behaviour is ok, because we only test test writing and reading from/to sqlite.
+    return rcutils_snprintf(
+      reinterpret_cast<char *>(buffer),
+      buffer_capacity,
+      "%c%c%c%c%c%c%c%c%s",
+      0x00, 0x01, 0x00, 0x00, 0x0f, 0x00, 0x00, 0x00,
+      message.c_str());
+  }
+
+  int get_buffer_capacity(const std::string & message)
+  {
+    return write_data_to_serialized_string_message(nullptr, 0, message);
+  }
+
+  std::shared_ptr<rcutils_uint8_array_t> make_serialized_message(std::string message)
```

Comment:
Maybe use the one from `rosbag2_storage::ros_helper` if possible?

Reply:
That's a good point, it serializes a little differently, but it'll do for the purpose of this test.